ui/apps/dev-server-ui/src/app/(dashboard)/stream/StreamDetails.tsx

```diff
@@ -53,6 +53,14 @@ export default function StreamDetails() {
     );
   }, [eventResult.data?.payload]);
 
+  if (eventResult.error || runResult.error) {
+    // If there's an error fetching the event or run, we should redirect to the
+    // stream. This happens a lot, since restarting the Dev Server will clear
+    // all data
+    router.push('/stream');
+    return null;
+  }
+
```

Comment:
If there is a real error, this ends up hiding it from the user right? I think this could be a good hotfix, but for long term, how could we implement this:

* An event is missing, redirect back to stream, w/ a toast that says "Event not found"
* An event has some other type of error response (5xx?), show the error in the event panel
* A run is missing, but the event is valid, show the error in the run panel
* A run fails to load, show the error in the run panel
* If both fail to load, redirect to stream and show a toast that says "Failed to load run"

I think it's more ideal DX to have more granular feedback for the user as some of these combinations are valid and could happen if there is some sort of failure in the backend. For now, I think this might be OK if we add a toast before `router.push()` that has copy like: 

```js
toast.message('Failed to load run data', {
  description: 'Please check the browser console for more info',
})
```